src/QueryFunction.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2020 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ElementHandle} from './JSHandle';
+
+export interface QueryFunction {
+  (element: Element, selector: string): (ElementHandle | null) | ElementHandle[];
+}
+
+const _customQueryFunctions = new Map<string, QueryFunction>();
```

Comment:
could we lose the `_` here? I think the fact it's not exported is enough.

Reply:
It chucks a lint error: `Individual declarations in merged declaration 'customQueryFunctions' must be all exported or all local.`

Reply:
It throws a lint error :(